File: backend/email-service.js
/**
 * 阿里云邮件发送服务
 * 用于发送用户反馈邮件
 */

const Dm20151123 = require('@alicloud/dm20151123');
const OpenApi = require('@alicloud/openapi-client');
const Util = require('@alicloud/tea-util');

class EmailService {
  constructor() {
    const config = new OpenApi.Config({
      accessKeyId: process.env.ALIYUN_AK_ID,
      accessKeySecret: process.env.ALIYUN_AK_SECRET,
    });
    config.endpoint = 'dm.aliyuncs.com';
    this.client = new Dm20151123.default(config);
  }

  /**
   * 发送单封邮件
   * @param {string} toAddress - 收件人邮箱
   * @param {string} subject - 邮件主题
   * @param {string} htmlBody - HTML邮件内容
   */
  async sendEmail(toAddress, subject, htmlBody) {
    const sendMailRequest = new Dm20151123.SingleSendMailRequest({
      accountName: 'bufftracker@mail.hezhili.online',
      addressType: 1,
      replyToAddress: true,
      toAddress: toAddress,
      subject: subject,
      htmlBody: htmlBody,
    });

    const runtime = new Util.RuntimeOptions({});
    const response = await this.client.singleSendMailWithOptions(sendMailRequest, runtime);
    return response;
  }

  /**
   * 发送用户反馈到作者邮箱
   * @param {string} feedback - 反馈内容
   * @param {string} userEmail - 用户邮箱（可选）
   */
  async sendFeedbackToAuthor(feedback, userEmail = '匿名用户') {
    const authorEmail = 'HEZH0014@e.ntu.edu.sg';
    const subject = `[网站反馈] 来自 ${userEmail} 的新反馈`;
    
    const htmlBody = `
      <!DOCTYPE html>
      <html>
      <head>
        <meta charset="utf-8">
        <style>
          body { font-family: 'Segoe UI', Tahoma, sans-serif; background: #1a1a2e; color: #eee; padding: 20px; }
          .container { max-width: 600px; margin: 0 auto; background: #16213e; border-radius: 10px; padding: 30px; }
          .header { color: #00ff88; font-size: 24px; margin-bottom: 20px; border-bottom: 2px solid #00ff88; padding-bottom: 10px; }
          .info { background: #0f3460; padding: 15px; border-radius: 8px; margin: 15px 0; }
          .label { color: #00ff88; font-weight: bold; }
          .content { background: #1a1a2e; padding: 20px; border-radius: 8px; margin-top: 15px; white-space: pre-wrap; line-height: 1.6; }
          .footer { margin-top: 20px; padding-top: 15px; border-top: 1px solid #333; color: #888; font-size: 12px; }
        </style>
      </head>
      <body>
        <div class="container">
          <div class="header">📬 新的网站反馈</div>
          <div class="info">
            <p><span class="label">发送者：</span>${userEmail}</p>
            <p><span class="label">时间：</span>${new Date().toLocaleString('zh-CN', { timeZone: 'Asia/Shanghai' })}</p>
          </div>
          <div class="label">反馈内容：</div>
          <div class="content">${feedback.replace(/\n/g, '<br>')}</div>
          <div class="footer">
            此邮件由 hezhili.online 自动发送
          </div>
        </div>
      </body>
      </html>
    `;

    return await this.sendEmail(authorEmail, subject, htmlBody);
  }

  /**
   * 发送确认邮件给用户
   * @param {string} userEmail - 用户邮箱
   * @param {string} feedbackPreview - 反馈内容预览
   */
  async sendConfirmationToUser(userEmail, feedbackPreview) {
    const subject = '感谢您的反馈 - hezhili.online';
    
    const htmlBody = `
      <!DOCTYPE html>
      <html>
      <head>
        <meta charset="utf-8">
        <style>
          body { font-family: 'Segoe UI', Tahoma, sans-serif; background: #1a1a2e; color: #eee; padding: 20px; }
          .container { max-width: 600px; margin: 0 auto; background: #16213e; border-radius: 10px; padding: 30px; }
          .header { color: #00ff88; font-size: 24px; margin-bottom: 20px; }
          .content { line-height: 1.8; }
          .highlight { background: #0f3460; padding: 15px; border-radius: 8px; margin: 20px 0; border-left: 4px solid #00ff88; }
          .preview { color: #aaa; font-style: italic; }
          .footer { margin-top: 30px; padding-top: 15px; border-top: 1px solid #333; color: #888; font-size: 12px; }
          a { color: #00ff88; }
        </style>
      </head>
      <body>
        <div class="container">
          <div class="header">✅ 反馈已收到</div>
          <div class="content">
            <p>您好！</p>
            <p>感谢您花时间向我提供反馈，您的意见对我非常重要。</p>
            
            <div class="highlight">
              <p><strong>您的反馈摘要：</strong></p>
              <p class="preview">${feedbackPreview.substring(0, 200)}${feedbackPreview.length > 200 ? '...' : ''}</p>
            </div>
            
            <p>我会认真阅读您的反馈，如有需要会尽快回复您。</p>
            <p>再次感谢！</p>
            <p style="margin-top: 20px;">— Zhili He</p>
          </div>
          <div class="footer">
            <p>此邮件由 <a href="https://hezhili.online">hezhili.online</a> 自动发送，请勿直接回复。</p>
          </div>
        </div>
      </body>
      </html>
    `;

    return await this.sendEmail(userEmail, subject, htmlBody);
  }
}

module.exports = EmailService;

File: backend/email_service.py
import aiosmtplib
from email.mime.text import MIMEText
from email.mime.multipart import MIMEMultipart
import os
from typing import Optional

class EmailService:
    def __init__(self):
        self.smtp_server = os.getenv("SMTP_SERVER", "smtp.aliyun.com")
        self.smtp_port = int(os.getenv("SMTP_PORT", "465"))
        self.username = os.getenv("SMTP_USERNAME")
        self.password = os.getenv("SMTP_PASSWORD")
        self.from_email = os.getenv("FROM_EMAIL", "bufftracker@mail.hezhili.online")

    async def send_email(self, to_email: str, subject: str, html_body: str):
        msg = MIMEMultipart("alternative")
        msg["Subject"] = subject
        msg["From"] = self.from_email
        msg["To"] = to_email

        html_part = MIMEText(html_body, "html")
        msg.attach(html_part)

        await aiosmtplib.send(
            msg,
            hostname=self.smtp_server,
            port=self.smtp_port,
            username=self.username,
            password=self.password,
            use_tls=True,
        )

    async def send_feedback_to_author(self, feedback: str, user_email: str = "匿名用户"):
        author_email = os.getenv("AUTHOR_EMAIL", "HEZH0014@e.ntu.edu.sg")
        subject = f"📬 新反馈 | 来自 {user_email}"
        timestamp = "2024-01-01 00:00:00"  # Use datetime.now() in real implementation

        html_body = f"""
        <!DOCTYPE html>
        <html>
        <head>
            <meta charset="utf-8">
            <style>
                body {{
                    font-family: 'Fira Code', 'Consolas', monospace;
                    background: #0a0a0a;
                    color: #00ff88;
                    padding: 20px;
                }}
                .terminal {{
                    max-width: 700px;
                    margin: 20px auto;
                    background: #0d0d0d;
                    border: 1px solid #00ff88;
                    border-radius: 8px;
                    padding: 20px;
                }}
            </style>
        </head>
        <body>
            <div class="terminal">
                <h2>🔔 新用户反馈</h2>
                <p><strong>时间:</strong> {timestamp}</p>
                <p><strong>用户:</strong> {user_email}</p>
                <p><strong>反馈内容:</strong></p>
                <pre>{feedback}</pre>
            </div>
        </body>
        </html>
        """

        await self.send_email(author_email, subject, html_body)

    async def send_confirmation_to_user(self, user_email: str, feedback: str):
        subject = "感谢您的反馈 - HeZhili"
        html_body = f"""
        <!DOCTYPE html>
        <html>
        <head>
            <meta charset="utf-8">
        </head>
        <body>
            <h1>感谢您的反馈！</h1>
            <p>我们已收到您的反馈：</p>
            <blockquote>{feedback[:200]}...</blockquote>
            <p>我们会尽快处理您的建议。</p>
        </body>
        </html>
        """

        await self.send_email(user_email, subject, html_body)